src/index.js

```diff
@@ -2,56 +2,147 @@ import F from 'futil'
 import _ from 'lodash/fp'
 import { observable, extendObservable, toJS } from 'mobx'
 import * as validators from './validators'
+import { buildPath, tokenizePath, joinPaths, pickFields } from './util'
 
 let unmerge = _.flow(F.simpleDiff, _.mapValues('to'))
 let changed = (x, y) => !_.isEqual(x, y) && !(F.isBlank(x) && F.isBlank(y))
 let Command = F.aspects.command(x => y => extendObservable(y, x))
+let getOrValue = (path, value) => _.getOr(value, path, value)
+let buildFieldPath = _.flow(tokenizePath, F.intersperse('fields'), _.compact)
 
-let values = _.mapValues('value')
 let Form = ({
   afterInitField = x => x,
   validate = validators.functions,
   ...config
 }) => {
-  let initField = (x, field) => {
+  let fieldMethods = (x, path = []) => {
     let node = observable({
-      field,
-      label: x.label || _.startCase(field),
-      // in case no value is provided, avoids controlled vs uncontrolled warning
-      // and need for mobx 4
-      value: _.getOr('', 'value', x),
-      get errors() {
-        return form.errors[field] || []
-      },
       get isValid() {
-        return !node.errors.length
+        return _.isEmpty(node.errors)
       },
       get isDirty() {
-        return changed(toJS(node.value), toJS(x.value))
+        return changed(
+          toJS(getOrValue(path, baseNode.value)),
+          toJS(getOrValue(path, form.value))
+        )
       },
+      getField: x => _.get(joinPaths(buildFieldPath(x)), node.fields),
       reset() {
-        node.value = F.when(_.isUndefined, '')(x.value)
+        if (_.isEmpty(path)) form.submit.state.error = null // Lil hack
```

Comment:
just a reminder about our call, reset only needs to set `values` and not `fields`